### PR TITLE
Fixes #17990: use toast notifications like foreman.

### PR DESCRIPTION
**Pull request by waldenraines**
Instead of using inline notifications we should be using toast
notifications like foreman does in order to match their style and UX.

http://projects.theforeman.org/issues/17990

### CODE REVIEW

**Comment by waldenraines** [comment]
Requires #171

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
Rework this to use https://github.com/theforeman/foreman/pull/4007

**Comment by ehelms** [comment]
Looks like this is now waiting on https://github.com/theforeman/foreman/pull/4300

**Comment by ehelms** [comment]
Looks like you can finish this now -- https://github.com/theforeman/foreman/pull/4469

**Comment by waldenraines** [comment]
> Looks like you can finish this now -- theforeman/foreman#4469

Yeah I know, will try to get around to it soon.

**Comment by ohadlevy** [comment]
any chance we can use this as an opportunity to hook into redux from angular ? (didnt check but quick google showed https://github.com/angular-redux/ng-redux)? I'm assuming you would need to use the new webpack support to import the current redux actions.

**Comment by waldenraines** [comment]
> any chance we can use this as an opportunity to hook into redux from angular ? (didnt check but quick google showed https://github.com/angular-redux/ng-redux)? I'm assuming you would need to use the new webpack support to import the current redux actions.

I was thinking the same thing the other day.  I will definitely take a look at doing this whenever I have a chance to revisit this PR.

**Comment by waldenraines** [comment]
This PR is now ready to be reviewed.  I used foreman's exposed notifications via `tfm.toastNotifications` but I did not wire this into react as it would have been too much work to justify throwing away.

File: app/assets/javascripts/bastion/components/notification.service.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        if (context) {
+            result = $interpolate(message)(context);
+        }
+
+        return result;
+    }
+
+    this.setSuccessMessage = function (message, context) {
+        foremanReact.toastNotifications.notify({message: interpolateIfNeeded(message, context), type: 'success'});
```
Comment (ohadlevy) [inline app/assets/javascripts/bastion/components/notification.service.js]: nitpick: but this doesnt actually go to the "react/redux" interface vs to the "legacy" tfm.toastNotifications name same, hence maybe it should not be called react. (to confuse other future readers).

Comment (waldenraines) [reply app/assets/javascripts/bastion/components/notification.service.js]: That's a good point, will update.

**Pull request merged**